pkg/runtime/runtime.go

```diff
@@ -54,17 +57,24 @@ type TemplateSpec struct {
 	// ObjApply is ApplyConfiguration for the TrainingRuntimes Template field.
 	ObjApply any
 	// PodSets is a set of Pod extracted from ObjApply.
+	// This is abstract concept to represent multiple PodSpec as a unit.
 	PodSets []PodSet
 }
 
 type PodSet struct {
+	// PodSet name is the name to identify PodSpec.
+	// This typically has the name stored in each PodSpec.
 	Name string
 	// If Name is trainer-node, CountForNonTrainer is null.
 	// For Trainer, PodSet Count should be stored in Info.RuntimePolicy.MLPolicy.NumNodes.
 	CountForNonTrainer *int32
+	InitContainers     []Container
```

Comment:
Should we add InitContainers once we have a use-case when InitContainer can be configured via TrainJob ?

Reply:
We want to keep InitContainers here since this is abstraction for Runtime.
So, each plugin can easily take and modify initContainers throughout cycle.

This is mostly preparation for overriding order changes (Plugin Change -> TrainJob -> Runtime).

Reply:
Sounds good.